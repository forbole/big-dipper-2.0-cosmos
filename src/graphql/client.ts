import WebSocket from 'isomorphic-ws';
import {
  ApolloClient,
  InMemoryCache,
  split,
  HttpLink,
  ApolloLink,
  concat,
} from '@apollo/client';
import {
  getMainDefinition,
} from '@apollo/client/utilities';
import { WebSocketLink } from '@apollo/client/link/ws';

import { useMemo } from 'react';

const defaultOptions:any = {
  watchQuery: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
  query: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
};

let apolloClient;

const httpLink = new HttpLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_URL,
});

const wsLink = new WebSocketLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_WS ?? 'wss://localhost:3000',
  options: {
    reconnect: true,
  },
  webSocketImpl: WebSocket,
});

const link = typeof window !== 'undefined' ? split(
  ({ query }) => {
    const {
      kind, operation,
    }:any = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'subscription';
  },
  wsLink,
  httpLink,
) : httpLink;

const authMiddleware = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: {
    },
  });

  return forward(operation);
});

function createApolloClient() {
  const client = new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: concat(authMiddleware, link),
    cache: new InMemoryCache({
    }),
  });

  client.defaultOptions = defaultOptions;

  return client;
}

export function initializeApollo(initialState = null) {
  // eslint-disable-next-line
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({
      ...existingCache, ...initialState,
    });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
