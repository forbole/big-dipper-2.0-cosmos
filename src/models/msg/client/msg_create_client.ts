import { Categories } from '../types';

class MsgCreateClient {
    public category: Categories;
    public type: string;
    public signer: string;
    public chainId: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.chainId = payload.chainId;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgCreateClient({
        json,
        type: json['@type'],
        signer: json.signer,
        chainId: json.client_state?.chain_id,
      });
    }
}

export default MsgCreateClient;
