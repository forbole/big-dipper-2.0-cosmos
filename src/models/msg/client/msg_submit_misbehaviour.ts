import { Categories } from '../types';

class MsgSubmitMisbehaviour {
    public category: Categories;
    public type: string;
    public signer: string;
    public clientId: string;
    public misbehaviour: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.clientId = payload.clientId;
      this.misbehaviour = payload.misbehaviour;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgSubmitMisbehaviour({
        json,
        type: json['@type'],
        signer: json.signer,
        clientId: json.client_id,
        misbehaviour: json.misbehaviour,
      });
    }
}

export default MsgSubmitMisbehaviour;
