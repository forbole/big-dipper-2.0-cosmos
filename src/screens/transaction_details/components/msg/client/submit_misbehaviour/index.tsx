import React from 'react';
import Trans from 'next-translate/Trans';
import { Typography } from '@material-ui/core';
import { Name } from '@components';
import { MsgSubmitMisbehaviour } from '@models';
import { useChainContext } from '@contexts';

const SubmitMisbehaviour = (props: {
    message: MsgSubmitMisbehaviour;
}) => {
  const { findAddress } = useChainContext();
  const { message } = props;

  const signer = findAddress(message.signer);
  const signerMoniker = signer ? signer?.moniker : message.signer;

  return (
    <Typography>
      <Trans
        i18nKey="message_contents:txSubmitMisbehaviourContent"
        components={[
          (
            <Name
              address={message.signer}
              name={signerMoniker}
            />
          ),
          <b />,
        ]}
        values={{
          clientId: message.clientId,
        }}
      />
    </Typography>
  );
};

export default SubmitMisbehaviour;
