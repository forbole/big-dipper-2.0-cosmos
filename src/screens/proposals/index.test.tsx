import React from 'react';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';

import { ApolloProvider } from '@apollo/client';
import { createMockClient } from 'mock-apollo-client';
import {
  ProposalsDocument,
} from '@graphql/types/general_types';
import Proposals from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
}));

jest.mock('./components', () => ({
  List: (props) => <div id="List" {...props} />,
}));

const mockProposals = jest.fn().mockResolvedValue({
  data: {
    proposals: [
      {
        title: 'Staking Param Change Part Two',
        proposalId: 7,
        status: 'PROPOSAL_STATUS_REJECTED',
        description: 'Update max validators',
      },
      {
        title: 'upgrading the software',
        proposalId: 6,
        status: 'PROPOSAL_STATUS_INVALID',
        description: 'software upgrade description',
      },
      {
        title: 'Staking Param Change',
        proposalId: 5,
        status: 'PROPOSAL_STATUS_INVALID',
        description: 'Update max validators',
      },
      {
        title: 'BD Proposal',
        proposalId: 4,
        status: 'PROPOSAL_STATUS_REJECTED',
        description: 'My not so awesome proposal',
      },
    ],
    total: {
      aggregate: {
        count: 4,
      },
    },
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Proposals', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();

    mockClient.setRequestHandler(
      ProposalsDocument,
      mockProposals,
    );
    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <Proposals />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
