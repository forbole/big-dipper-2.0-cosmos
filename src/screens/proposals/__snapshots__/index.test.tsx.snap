// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Proposals matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Layout"
  navTitle="proposals:proposals"
>
  <div
    id="List"
    isItemLoaded={[Function]}
    itemCount={4}
    items={
      Array [
        Object {
          "description": "Update max validators",
          "id": 7,
          "status": "PROPOSAL_STATUS_REJECTED",
          "title": "Staking Param Change Part Two",
        },
        Object {
          "description": "software upgrade description",
          "id": 6,
          "status": "PROPOSAL_STATUS_INVALID",
          "title": "upgrading the software",
        },
        Object {
          "description": "Update max validators",
          "id": 5,
          "status": "PROPOSAL_STATUS_INVALID",
          "title": "Staking Param Change",
        },
        Object {
          "description": "My not so awesome proposal",
          "id": 4,
          "status": "PROPOSAL_STATUS_REJECTED",
          "title": "BD Proposal",
        },
      ]
    }
    loadMoreItems={[Function]}
    rawDataTotal={4}
  />
</div>
`;
